src/Compilers/CSharp/Portable/Compiler/Instrumentation.cs

```diff
@@ -203,5 +238,43 @@ private BoundNode CollectDynamicAnalysis(BoundStatement statement)
             
             return statementFactory.Block(ImmutableArray.Create(cellAssignment, statement));
         }
+
+        private static CSharpSyntaxNode SyntaxForSpan(BoundStatement statement)
+        {
+            CSharpSyntaxNode syntaxForSpan;
+
+            switch (statement.Kind)
+            {
+                case BoundKind.IfStatement:
+                    syntaxForSpan = ((BoundIfStatement)statement).Condition.Syntax;
+                    break;
+                case BoundKind.WhileStatement:
+                    syntaxForSpan = ((BoundWhileStatement)statement).Condition.Syntax;
+                    break;
+                case BoundKind.ForEachStatement:
+                    syntaxForSpan = ((BoundForEachStatement)statement).Expression.Syntax;
+                    break;
+                case BoundKind.DoStatement:
+                    syntaxForSpan = ((BoundDoStatement)statement).Condition.Syntax;
+                    break;
+                case BoundKind.UsingStatement:
+                    {
+                        BoundUsingStatement usingStatement = (BoundUsingStatement)statement;
+                        syntaxForSpan = ((BoundNode)usingStatement.ExpressionOpt ?? usingStatement.DeclarationsOpt).Syntax;
+                        break;
+                    }
+                case BoundKind.LockStatement:
+                    syntaxForSpan = ((BoundLockStatement)statement).Argument.Syntax;
+                    break;
+                case BoundKind.SwitchStatement:
+                    syntaxForSpan = ((BoundSwitchStatement)statement).Expression.Syntax;
+                    break;
+                default:
+                    syntaxForSpan = statement.Syntax;
+                    break;
+            }
+
+            return syntaxForSpan;
```

Comment:
We had tests verifying that the set of created instrumentation slots was as expected. I have added a test to verify that the spans are as expected.